packages/core/test/tag/compoundTagTests.tsx

```diff
@@ -33,13 +34,24 @@ describe("<CompoundTag>", () => {
 
     it("renders icons", () => {
         const wrapper = shallow(
-            <CompoundTag icon="tick" rightIcon="airplane" leftContent="Hello">
+            <CompoundTag icon="tick" endIcon="airplane" leftContent="Hello">
                 World
             </CompoundTag>,
         );
         assert.lengthOf(wrapper.find(Icon), 2);
     });
 
+    it("prefers endIcon to rightIcon", () => {
+        const result = render(
+            // eslint-disable-next-line @typescript-eslint/no-deprecated
+            <CompoundTag endIcon="airplane" rightIcon="tick" leftContent="Hello">
+                World
+            </CompoundTag>,
+        );
+        expect(result.baseElement.querySelector(`.${Classes.ICON}-airplane`)).to.exist;
+        expect(result.baseElement.querySelector(`.${Classes.ICON}-add`)).not.to.exist;
```

Comment:
The test setup above renders `rightIcon="tick"`, but we're testing for the absence of the "add" icon. Should this be:
```suggestion
        expect(result.baseElement.querySelector(`.${Classes.ICON}-tick`)).not.to.exist;
```